### PR TITLE
Added: Documentation for Serializers

### PR DESCRIPTION
**Pull request by divyanshugit**
This PR resolves issue #59.

- [x] Added the inline documentation for Serializers.

### CODE REVIEW

File: fedrec/serialization/serializers.py [hunk 1]
```diff
@@ -86,15 +180,49 @@ def deserialize(cls, obj):
...
+    Attributes:
+    ----------
+    serializer: str
+        The serializer to use.
+    """
 
     @classmethod
     def serialize(cls, obj):
+        """
+        Serializes an JSON object.
```
Comment (vkkhare) [inline fedrec/serialization/serializers.py:195]: Serializes a python object to json

File: fedrec/serialization/serializers.py [hunk 2]
```diff
@@ -86,15 +180,49 @@ def deserialize(cls, obj):
 
 @registry.load("serializer", "json")
 class JSONSerializer(AbstractSerializer):
+    """
+    JSONSerializer serializes and deserializes json objects.
```
Comment (vkkhare) [inline fedrec/serialization/serializers.py:184]: Uses json serialization strategy for objects

**Review by vkkhare — CHANGES_REQUESTED** [review]

**Review by ramesht007 — COMMENTED** [review]
In Json deserialize I think we can also add

Deserializes a JSON object and adds to registry based on `__type__` key.

File: fedrec/serialization/serializers.py [hunk 3]
```diff
@@ -14,9 +14,37 @@
...
+    serialize(obj):
+        Serializes an object.
+    deserialize(obj):
+        Deserializes an object.
+    """
 
     @classmethod
     def generate_message_dict(cls, obj):
+        """
+        Generates a dictionary from an object.
```
Comment (vkkhare) [inline fedrec/serialization/serializers.py:36]: appends type information for finding the appropriate serialiser

File: fedrec/serialization/serializers.py [hunk 4]
```diff
@@ -86,15 +180,49 @@ def deserialize(cls, obj):
...
+            The serialized object.
+        """
         obj = cls.generate_message_dict(obj)
         print(obj, type(obj))
         return json.dumps(obj, indent=4).encode('utf-8')
 
     @classmethod
     def deserialize(cls, obj):
+        """
+        Deserializes a JSON object and adds to registry based on `__type__` key.
```
Comment (vkkhare) [inline fedrec/serialization/serializers.py:213]: It does not add to registry rather fetches the correct serializer from registry based on __type__

**Review by vkkhare — CHANGES_REQUESTED** [review]

**Review by vkkhare — APPROVED** [review]

**Comment by ramesht007** [comment]
Well done @divyanshugit !!

**Pull request merged**